### DISCUSSION TITLE
how can i override the code block in autogenerated APIPage component

### DISCUSSION THREAD

**Discussion by ryanwaits** [fuma-nama/fumadocs#1108] [Q&A] [answered]
I have a custom code block component that I am using for all code snippets and I'd like to upgrade to the `fumadocs-openapi` package to use the `APIPage` component, but the code blocks use the default code block and themes.

Is there a way to override this? Or are there any plans to provide that ability?

**Comment by fuma-nama** [comment]
You can change Shiki options from `createOpenAPI`

**Reply by ryanwaits** [reply]
it looks like some of the other components that get rendered with `<APIPage>` are not using the `shikiOptions` set in `createOpenAPI` for setting a custom theme 

Code:

```ts
export const openapi = createOpenAPI({
  shikiOptions: {
    themes: {
      dark: customTheme.dark,
      light: customTheme.light
    },
  },
});
```

For instance, the custom theme is working for the `CodeExample` and not for others: https://github.com/fuma-nama/fumadocs/blob/fff3211b98f11c71f3abde7e008fcbd80fee2565/packages/openapi/src/ui/contexts/code-example.tsx#L114

<img width="456" alt="Screenshot 2025-03-14 at 10 10 16 AM" src="https://github.com/user-attachments/assets/5c53bed3-65de-4ca2-9576-b26fea4cd179" />
<img width="589" alt="Screenshot 2025-03-14 at 10 30 37 AM" src="https://github.com/user-attachments/assets/f131bed3-17cd-4adb-94da-de99c912e34d" />

Maybe this is something the `@fumadocs/cli` could add for more customization?

**Reply by fuma-nama** [reply]
Thanks for reporting, I'll fix it

**Reply by ryanwaits** [reply]
any updates here?

**Reply by fuma-nama** [reply, marked as answer]
Fixed long time ago, have you upgraded Fumadocs?

**Reply by ryanwaits** [reply]
fantastic, just updated and it looks good - appreciate it as always!
